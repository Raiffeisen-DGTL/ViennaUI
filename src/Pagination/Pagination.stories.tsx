import React from 'react';
import { Story, Meta } from 'storybook';
import { Pagination, PaginationProps } from './Pagination';

export default {
    title: 'Development/Pagination',
    component: Pagination,
} as Meta;

export const Overview: Story<PaginationProps> = (args) => {
    return <Pagination pageSize={25} totalItemsCount={500} onChange={() => ''} {...args} />;
};
