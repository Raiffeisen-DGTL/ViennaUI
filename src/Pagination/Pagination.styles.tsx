import styled, { css } from 'styled-components';
import { pagination } from 'vienna.ui-theme';
import { getPresets } from '../Utils/styling';

const presets = getPresets(
    pagination,
    'pagination'
)({
    gap: '$size',
});

export interface PropsBox {
    $size?: 'm' | 'l' | 's';
    $align?: 'flex-start' | 'center' | 'flex-end' | 'space-between' | 'space-around';
}
export const Box = styled.div<PropsBox>`
    display: inline-flex;
    flex-direction: row;
    align-content: flex-start;
    flex-wrap: wrap;
    vertical-align: middle;
    width: 100%;

    ${({ $align }) =>
        $align &&
        css`
            justify-content: ${$align};
        `}

    ${presets.gap}
`;
