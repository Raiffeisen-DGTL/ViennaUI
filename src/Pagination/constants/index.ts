export const ELLIPSIS_PREV = -1;
export const ELLIPSIS_NEXT = -2;
export const ARROW_PREV = -3;
export const ARROW_NEXT = -4;
