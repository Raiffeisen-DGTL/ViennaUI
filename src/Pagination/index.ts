export { Pagination, type PaginationProps } from './Pagination';
