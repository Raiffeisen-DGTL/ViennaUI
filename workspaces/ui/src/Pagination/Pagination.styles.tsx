import styled, { css } from 'styled-components';
import { getPresets } from 'vienna.ui-primitives';

const presets = getPresets('pagination', {
    gap: 'size',
});

export const Box = styled.div<any>`
    display: inline-flex;
    flex-direction: row;
    align-content: flex-start;
    flex-wrap: wrap;
    vertical-align: middle;
    width: 100%;

    ${({ align }) =>
        align &&
        css`
            justify-content: ${align};
        `}

    ${presets.gap}
`;
