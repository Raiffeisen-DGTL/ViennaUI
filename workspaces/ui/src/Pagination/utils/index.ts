export * from './range';
export * from './getLastIndex';
export * from './getPagersNumbers';
