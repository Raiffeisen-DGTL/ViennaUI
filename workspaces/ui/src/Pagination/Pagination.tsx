import React, { useState, useCallback, useMemo, useEffect } from 'react';
import { GoLeft, GoRight } from 'vienna.icons';
import { getLastIndex, getPagersNumbers } from './utils';
import { Pager } from './Pager';
import { Box } from './Pagination.styles';

const ELLIPSIS_PREV = -1;
const ELLIPSIS_NEXT = -2;

interface Props {
    /**
     * Размер компонента
     */
    size?: 'm' | 'l' | 's';
    /**
     * Выравнивание по горизонтали
     */
    align?: 'flex-start' | 'center' | 'flex-end' | 'space-between' | 'space-around';
    /**
     * Количество элементов на одной странице
     */
    pageSize: number;
    /**
     * Индекс текущей страницы (начинается с 0)
     */
    initialPageIndex?: number;
    /**
     * Общее количество элементов
     */
    totalItemsCount: number;
    /**
     * Количество кнопок-пэйджеров, которые нужно рендерить по соседству от текущего пэйджера,
     * когда показываются кнопки с эллипсисами (jumpers)
     */
    currentPageNeighboursCount?: number;
    /**
     * Коллбэк. Вызывается при изменении страницы
     * В этом методе нужно описывать логику показа элементов в зависимости
     * от страницы и размера страницы
     * @param pageIndex - индекс новой страницы
     * @param pageSize - количество элементов на странице
     */
    onChange: (event: React.FormEvent | null, data: { pageIndex: number; pageSize: number }) => void;
}

interface HTMLAttributeProps {
    [key: string]: any;
    id?: string;
    title?: string;
    className?: string;
    role?: string;
    spellCheck?: boolean;
    tabIndex?: number;
}

export type PaginationProps = Props & HTMLAttributeProps;

export const Pagination: React.FC<PaginationProps> = (props): JSX.Element => {
    const {
        size = 'm',
        initialPageIndex,
        pageSize,
        totalItemsCount,
        onChange,
        currentPageNeighboursCount = 2,
        ...attrs
    } = props;

    const lastPageIndex = useMemo(() => {
        return getLastIndex(totalItemsCount, pageSize);
    }, [totalItemsCount, pageSize]);
    const [page, setPage] = useState(initialPageIndex && initialPageIndex <= lastPageIndex ? initialPageIndex : 0);

    const handleSelectPage = useCallback(
        (event: React.FormEvent | null, pageIndex: number) => {
            setPage(pageIndex);
            onChange(event, { pageIndex, pageSize });
        },
        [onChange, pageSize, setPage]
    );

    useEffect(() => {
        setPage(page);
        onChange(null, { pageIndex: page, pageSize });
    }, [pageSize]);

    useEffect(() => {
        if (page > lastPageIndex) {
            handleSelectPage(null, lastPageIndex);
        }
    }, [lastPageIndex]);

    const pagersNumbers = useMemo(() => {
        return getPagersNumbers(page, lastPageIndex, currentPageNeighboursCount, ELLIPSIS_PREV, ELLIPSIS_NEXT);
    }, [currentPageNeighboursCount, page, lastPageIndex]);

    const handleSelectNext = useCallback(
        (event: React.FormEvent) => {
            const newCurrentIndex = page === lastPageIndex ? lastPageIndex : page + 1;
            handleSelectPage(event, newCurrentIndex);
        },
        [page, handleSelectPage, lastPageIndex]
    );

    const handleSelectPrev = useCallback(
        (event: React.FormEvent) => {
            const newCurrentIndex = page === 0 ? 0 : page - 1;
            handleSelectPage(event, newCurrentIndex);
        },
        [handleSelectPage, page]
    );

    const handleEllipsisBack = useCallback(
        (event: React.FormEvent) => {
            const newPageIndex = Math.ceil(page / 2);
            handleSelectPage(event, newPageIndex);
        },
        [page, handleSelectPage]
    );

    const handleEllipsisNext = useCallback(
        (event: React.FormEvent) => {
            const newPageIndex = Math.ceil((lastPageIndex + page) / 2);
            handleSelectPage(event, newPageIndex);
        },
        [page, handleSelectPage, lastPageIndex]
    );

    const renderPagers = useCallback((): React.ReactNode[] => {
        return pagersNumbers.map(
            (index: number): React.ReactNode => {
                if (index === ELLIPSIS_PREV || index === ELLIPSIS_NEXT) {
                    const clickHandler = index === ELLIPSIS_PREV ? handleEllipsisBack : handleEllipsisNext;
                    return (
                        <Pager key={index.toString()} tabIndex={1} active={false} size={size} onClick={clickHandler}>
                            ...
                        </Pager>
                    );
                }
                const onClick = (event: React.FormEvent) => handleSelectPage(event, index);
                return (
                    <Pager key={index.toString()} tabIndex={1} active={index === page} size={size} onClick={onClick}>
                        {index + 1}
                    </Pager>
                );
            }
        );
    }, [page, size, pagersNumbers, handleEllipsisNext, handleEllipsisBack, handleSelectPage]);

    return (
        <Box {...attrs}>
            <Pager disabled={page === 0} tabIndex={1} size={size} onClick={handleSelectPrev}>
                <GoLeft size={size} />
            </Pager>
            {renderPagers()}
            <Pager disabled={page >= lastPageIndex} tabIndex={1} size={size} onClick={handleSelectNext}>
                <GoRight size={size} />
            </Pager>
        </Box>
    );
};

Pagination.displayName = 'Pagination';
export default Pagination;
