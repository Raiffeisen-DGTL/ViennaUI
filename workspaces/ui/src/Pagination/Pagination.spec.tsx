import React from 'react';
import Pagination, { PaginationProps } from './Pagination';

test('Pagination', () => {
    const snap = snapshot.render(<Pagination pageSize={25} totalItemsCount={100} onChange={jest.fn()} />);
    expect(snap).toMatchSnapshot();
});

test('Pagination w/ initialPageIndex', () => {
    const snap = snapshot.render(
        <Pagination initialPageIndex={3} pageSize={25} totalItemsCount={100} onChange={jest.fn()} />
    );

    expect(snap).toMatchSnapshot();
});

test('Pagination w/ currentPageNeighboursCount', () => {
    const snap = snapshot.render(
        <Pagination currentPageNeighboursCount={3} pageSize={25} totalItemsCount={100} onChange={jest.fn()} />
    );

    expect(snap).toMatchSnapshot();
});

test('Pagination w/ size', () => {
    const sizes: PaginationProps['size'][] = ['s', 'm', 'l'];

    sizes.forEach((size) => {
        const snap = snapshot.render(
            <Pagination size={size} pageSize={25} totalItemsCount={100} onChange={jest.fn()} />
        );

        expect(snap).toMatchSnapshot();
    });
});

test('Pagination w/ onChange (next)', () => {
    const fn = jest.fn();
    const event = { currentTarget: {} };
    const snap = snapshot.shallow(<Pagination pageSize={25} totalItemsCount={100} onChange={fn} />);

    snap.children().last().simulate('click', event);

    expect(snap).toMatchSnapshot();
    expect(fn).toHaveBeenCalled();
    expect(fn).toHaveBeenCalledWith(event, { pageIndex: 1, pageSize: 25 });

    expect(snap).toMatchSnapshot();
});

test('Pagination w/ onChange (target)', () => {
    const fn = jest.fn();
    const event = { currentTarget: {} };
    const snap = snapshot.shallow(<Pagination pageSize={25} totalItemsCount={100} onChange={fn} />);

    snap.childAt(2).simulate('click', event);

    expect(snap).toMatchSnapshot();
    expect(fn).toHaveBeenCalled();
    expect(fn).toHaveBeenCalledWith(event, { pageIndex: 1, pageSize: 25 });

    expect(snap).toMatchSnapshot();
});

test('Pagination w/ onChange (prev)', () => {
    const fn = jest.fn();
    const event = { currentTarget: {} };
    const snap = snapshot.shallow(
        <Pagination initialPageIndex={3} pageSize={25} totalItemsCount={100} onChange={fn} />
    );

    snap.children().first().simulate('click', event);

    expect(snap).toMatchSnapshot();
    expect(fn).toHaveBeenCalled();
    expect(fn).toHaveBeenCalledWith(event, { pageIndex: 2, pageSize: 25 });

    expect(snap).toMatchSnapshot();
});

test('Pagination w/ onChange (first and prev)', () => {
    const fn = jest.fn();
    const event = { currentTarget: {} };
    const snap = snapshot.shallow(<Pagination pageSize={25} totalItemsCount={100} onChange={fn} />);

    snap.children().first().simulate('click', event);

    expect(snap).toMatchSnapshot();
    expect(fn).toHaveBeenCalled();
    expect(fn).toHaveBeenCalledWith(event, { pageIndex: 0, pageSize: 25 });

    expect(snap).toMatchSnapshot();
});

test('Pagination w/ onChange (last and next)', () => {
    const fn = jest.fn();
    const event = { currentTarget: {} };
    const snap = snapshot.shallow(
        <Pagination initialPageIndex={3} pageSize={25} totalItemsCount={100} onChange={fn} />
    );

    snap.children().last().simulate('click', event);
    snap.children().last().simulate('click', event);

    expect(snap).toMatchSnapshot();
    expect(fn).toHaveBeenCalled();
    expect(fn).toHaveBeenCalledWith(event, { pageIndex: 3, pageSize: 25 });

    expect(snap).toMatchSnapshot();
});
