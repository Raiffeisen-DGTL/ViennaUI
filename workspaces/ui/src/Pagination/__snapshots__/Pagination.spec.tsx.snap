// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination 1`] = `
<div
  class="Paginationstyles__Box-sc-7a3gdv-0 kCPgsd"
>
  <button
    class="Pagerstyles__Box-sc-1weko6t-0 bVOtWq"
    disabled=""
    tabindex="-1"
  >
    <svg
      fill="currentColor"
      height="20"
      viewBox="0 0 24 24"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs />
      <path
        d="M9.56 11.854l7.294 7.293-1.415 1.414-8-8a1 1 0 010-1.415l8-8 1.415 1.415-7.293 7.293z"
      />
    </svg>
  </button>
  <button
    class="Pagerstyles__Box-sc-1weko6t-0 hDiJml"
    tabindex="1"
  >
    1
  </button>
  <button
    class="Pagerstyles__Box-sc-1weko6t-0 dWrcUk"
    tabindex="1"
  >
    2
  </button>
  <button
    class="Pagerstyles__Box-sc-1weko6t-0 dWrcUk"
    tabindex="1"
  >
    3
  </button>
  <button
    class="Pagerstyles__Box-sc-1weko6t-0 dWrcUk"
    tabindex="1"
  >
    4
  </button>
  <button
    class="Pagerstyles__Box-sc-1weko6t-0 dWrcUk"
    tabindex="1"
  >
    <svg
      fill="currentColor"
      height="20"
      viewBox="0 0 24 24"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs />
      <path
        d="M14.44 11.854l-7.294 7.293 1.415 1.414 8-8a1 1 0 000-1.415l-8-8-1.415 1.415 7.293 7.293z"
      />
    </svg>
  </button>
</div>
`;

exports[`Pagination w/ currentPageNeighboursCount 1`] = `
<div
  class="Paginationstyles__Box-sc-7a3gdv-0 kCPgsd"
>
  <button
    class="Pagerstyles__Box-sc-1weko6t-0 bVOtWq"
    disabled=""
    tabindex="-1"
  >
    <svg
      fill="currentColor"
      height="20"
      viewBox="0 0 24 24"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs />
      <path
        d="M9.56 11.854l7.294 7.293-1.415 1.414-8-8a1 1 0 010-1.415l8-8 1.415 1.415-7.293 7.293z"
      />
    </svg>
  </button>
  <button
    class="Pagerstyles__Box-sc-1weko6t-0 hDiJml"
    tabindex="1"
  >
    1
  </button>
  <button
    class="Pagerstyles__Box-sc-1weko6t-0 dWrcUk"
    tabindex="1"
  >
    2
  </button>
  <button
    class="Pagerstyles__Box-sc-1weko6t-0 dWrcUk"
    tabindex="1"
  >
    3
  </button>
  <button
    class="Pagerstyles__Box-sc-1weko6t-0 dWrcUk"
    tabindex="1"
  >
    4
  </button>
  <button
    class="Pagerstyles__Box-sc-1weko6t-0 dWrcUk"
    tabindex="1"
  >
    <svg
      fill="currentColor"
      height="20"
      viewBox="0 0 24 24"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs />
      <path
        d="M14.44 11.854l-7.294 7.293 1.415 1.414 8-8a1 1 0 000-1.415l-8-8-1.415 1.415 7.293 7.293z"
      />
    </svg>
  </button>
</div>
`;

exports[`Pagination w/ initialPageIndex 1`] = `
<div
  class="Paginationstyles__Box-sc-7a3gdv-0 kCPgsd"
>
  <button
    class="Pagerstyles__Box-sc-1weko6t-0 dWrcUk"
    tabindex="1"
  >
    <svg
      fill="currentColor"
      height="20"
      viewBox="0 0 24 24"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs />
      <path
        d="M9.56 11.854l7.294 7.293-1.415 1.414-8-8a1 1 0 010-1.415l8-8 1.415 1.415-7.293 7.293z"
      />
    </svg>
  </button>
  <button
    class="Pagerstyles__Box-sc-1weko6t-0 dWrcUk"
    tabindex="1"
  >
    1
  </button>
  <button
    class="Pagerstyles__Box-sc-1weko6t-0 dWrcUk"
    tabindex="1"
  >
    2
  </button>
  <button
    class="Pagerstyles__Box-sc-1weko6t-0 dWrcUk"
    tabindex="1"
  >
    3
  </button>
  <button
    class="Pagerstyles__Box-sc-1weko6t-0 hDiJml"
    tabindex="1"
  >
    4
  </button>
  <button
    class="Pagerstyles__Box-sc-1weko6t-0 bVOtWq"
    disabled=""
    tabindex="-1"
  >
    <svg
      fill="currentColor"
      height="20"
      viewBox="0 0 24 24"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs />
      <path
        d="M14.44 11.854l-7.294 7.293 1.415 1.414 8-8a1 1 0 000-1.415l-8-8-1.415 1.415 7.293 7.293z"
      />
    </svg>
  </button>
</div>
`;

exports[`Pagination w/ onChange (first and prev) 1`] = `
<Paginationstyles__Box>
  <Pager
    active={false}
    disabled={true}
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    <GoLeft
      color="currentColor"
      size="m"
    />
  </Pager>
  <Pager
    active={true}
    key="0"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    1
  </Pager>
  <Pager
    active={false}
    key="1"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    2
  </Pager>
  <Pager
    active={false}
    key="2"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    3
  </Pager>
  <Pager
    active={false}
    key="3"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    4
  </Pager>
  <Pager
    active={false}
    disabled={false}
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    <GoRight
      color="currentColor"
      size="m"
    />
  </Pager>
</Paginationstyles__Box>
`;

exports[`Pagination w/ onChange (first and prev) 2`] = `
<Paginationstyles__Box>
  <Pager
    active={false}
    disabled={true}
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    <GoLeft
      color="currentColor"
      size="m"
    />
  </Pager>
  <Pager
    active={true}
    key="0"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    1
  </Pager>
  <Pager
    active={false}
    key="1"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    2
  </Pager>
  <Pager
    active={false}
    key="2"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    3
  </Pager>
  <Pager
    active={false}
    key="3"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    4
  </Pager>
  <Pager
    active={false}
    disabled={false}
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    <GoRight
      color="currentColor"
      size="m"
    />
  </Pager>
</Paginationstyles__Box>
`;

exports[`Pagination w/ onChange (last and next) 1`] = `
<Paginationstyles__Box>
  <Pager
    active={false}
    disabled={false}
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    <GoLeft
      color="currentColor"
      size="m"
    />
  </Pager>
  <Pager
    active={false}
    key="0"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    1
  </Pager>
  <Pager
    active={false}
    key="1"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    2
  </Pager>
  <Pager
    active={false}
    key="2"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    3
  </Pager>
  <Pager
    active={true}
    key="3"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    4
  </Pager>
  <Pager
    active={false}
    disabled={true}
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    <GoRight
      color="currentColor"
      size="m"
    />
  </Pager>
</Paginationstyles__Box>
`;

exports[`Pagination w/ onChange (last and next) 2`] = `
<Paginationstyles__Box>
  <Pager
    active={false}
    disabled={false}
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    <GoLeft
      color="currentColor"
      size="m"
    />
  </Pager>
  <Pager
    active={false}
    key="0"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    1
  </Pager>
  <Pager
    active={false}
    key="1"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    2
  </Pager>
  <Pager
    active={false}
    key="2"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    3
  </Pager>
  <Pager
    active={true}
    key="3"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    4
  </Pager>
  <Pager
    active={false}
    disabled={true}
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    <GoRight
      color="currentColor"
      size="m"
    />
  </Pager>
</Paginationstyles__Box>
`;

exports[`Pagination w/ onChange (next) 1`] = `
<Paginationstyles__Box>
  <Pager
    active={false}
    disabled={false}
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    <GoLeft
      color="currentColor"
      size="m"
    />
  </Pager>
  <Pager
    active={false}
    key="0"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    1
  </Pager>
  <Pager
    active={true}
    key="1"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    2
  </Pager>
  <Pager
    active={false}
    key="2"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    3
  </Pager>
  <Pager
    active={false}
    key="3"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    4
  </Pager>
  <Pager
    active={false}
    disabled={false}
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    <GoRight
      color="currentColor"
      size="m"
    />
  </Pager>
</Paginationstyles__Box>
`;

exports[`Pagination w/ onChange (next) 2`] = `
<Paginationstyles__Box>
  <Pager
    active={false}
    disabled={false}
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    <GoLeft
      color="currentColor"
      size="m"
    />
  </Pager>
  <Pager
    active={false}
    key="0"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    1
  </Pager>
  <Pager
    active={true}
    key="1"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    2
  </Pager>
  <Pager
    active={false}
    key="2"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    3
  </Pager>
  <Pager
    active={false}
    key="3"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    4
  </Pager>
  <Pager
    active={false}
    disabled={false}
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    <GoRight
      color="currentColor"
      size="m"
    />
  </Pager>
</Paginationstyles__Box>
`;

exports[`Pagination w/ onChange (prev) 1`] = `
<Paginationstyles__Box>
  <Pager
    active={false}
    disabled={false}
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    <GoLeft
      color="currentColor"
      size="m"
    />
  </Pager>
  <Pager
    active={false}
    key="0"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    1
  </Pager>
  <Pager
    active={false}
    key="1"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    2
  </Pager>
  <Pager
    active={true}
    key="2"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    3
  </Pager>
  <Pager
    active={false}
    key="3"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    4
  </Pager>
  <Pager
    active={false}
    disabled={false}
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    <GoRight
      color="currentColor"
      size="m"
    />
  </Pager>
</Paginationstyles__Box>
`;

exports[`Pagination w/ onChange (prev) 2`] = `
<Paginationstyles__Box>
  <Pager
    active={false}
    disabled={false}
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    <GoLeft
      color="currentColor"
      size="m"
    />
  </Pager>
  <Pager
    active={false}
    key="0"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    1
  </Pager>
  <Pager
    active={false}
    key="1"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    2
  </Pager>
  <Pager
    active={true}
    key="2"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    3
  </Pager>
  <Pager
    active={false}
    key="3"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    4
  </Pager>
  <Pager
    active={false}
    disabled={false}
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    <GoRight
      color="currentColor"
      size="m"
    />
  </Pager>
</Paginationstyles__Box>
`;

exports[`Pagination w/ onChange (target) 1`] = `
<Paginationstyles__Box>
  <Pager
    active={false}
    disabled={false}
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    <GoLeft
      color="currentColor"
      size="m"
    />
  </Pager>
  <Pager
    active={false}
    key="0"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    1
  </Pager>
  <Pager
    active={true}
    key="1"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    2
  </Pager>
  <Pager
    active={false}
    key="2"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    3
  </Pager>
  <Pager
    active={false}
    key="3"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    4
  </Pager>
  <Pager
    active={false}
    disabled={false}
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    <GoRight
      color="currentColor"
      size="m"
    />
  </Pager>
</Paginationstyles__Box>
`;

exports[`Pagination w/ onChange (target) 2`] = `
<Paginationstyles__Box>
  <Pager
    active={false}
    disabled={false}
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    <GoLeft
      color="currentColor"
      size="m"
    />
  </Pager>
  <Pager
    active={false}
    key="0"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    1
  </Pager>
  <Pager
    active={true}
    key="1"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    2
  </Pager>
  <Pager
    active={false}
    key="2"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    3
  </Pager>
  <Pager
    active={false}
    key="3"
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    4
  </Pager>
  <Pager
    active={false}
    disabled={false}
    onClick={[Function]}
    size="m"
    tabIndex={1}
  >
    <GoRight
      color="currentColor"
      size="m"
    />
  </Pager>
</Paginationstyles__Box>
`;

exports[`Pagination w/ size 1`] = `
<div
  class="Paginationstyles__Box-sc-7a3gdv-0 kCPgsd"
>
  <button
    class="Pagerstyles__Box-sc-1weko6t-0 gXWkbU"
    disabled=""
    tabindex="-1"
  >
    <svg
      fill="currentColor"
      height="16"
      viewBox="0 0 24 24"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs />
      <path
        d="M9.56 11.854l7.294 7.293-1.415 1.414-8-8a1 1 0 010-1.415l8-8 1.415 1.415-7.293 7.293z"
      />
    </svg>
  </button>
  <button
    class="Pagerstyles__Box-sc-1weko6t-0 qgcsH"
    tabindex="1"
  >
    1
  </button>
  <button
    class="Pagerstyles__Box-sc-1weko6t-0 csdZFe"
    tabindex="1"
  >
    2
  </button>
  <button
    class="Pagerstyles__Box-sc-1weko6t-0 csdZFe"
    tabindex="1"
  >
    3
  </button>
  <button
    class="Pagerstyles__Box-sc-1weko6t-0 csdZFe"
    tabindex="1"
  >
    4
  </button>
  <button
    class="Pagerstyles__Box-sc-1weko6t-0 csdZFe"
    tabindex="1"
  >
    <svg
      fill="currentColor"
      height="16"
      viewBox="0 0 24 24"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs />
      <path
        d="M14.44 11.854l-7.294 7.293 1.415 1.414 8-8a1 1 0 000-1.415l-8-8-1.415 1.415 7.293 7.293z"
      />
    </svg>
  </button>
</div>
`;

exports[`Pagination w/ size 2`] = `
<div
  class="Paginationstyles__Box-sc-7a3gdv-0 kCPgsd"
>
  <button
    class="Pagerstyles__Box-sc-1weko6t-0 bVOtWq"
    disabled=""
    tabindex="-1"
  >
    <svg
      fill="currentColor"
      height="20"
      viewBox="0 0 24 24"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs />
      <path
        d="M9.56 11.854l7.294 7.293-1.415 1.414-8-8a1 1 0 010-1.415l8-8 1.415 1.415-7.293 7.293z"
      />
    </svg>
  </button>
  <button
    class="Pagerstyles__Box-sc-1weko6t-0 hDiJml"
    tabindex="1"
  >
    1
  </button>
  <button
    class="Pagerstyles__Box-sc-1weko6t-0 dWrcUk"
    tabindex="1"
  >
    2
  </button>
  <button
    class="Pagerstyles__Box-sc-1weko6t-0 dWrcUk"
    tabindex="1"
  >
    3
  </button>
  <button
    class="Pagerstyles__Box-sc-1weko6t-0 dWrcUk"
    tabindex="1"
  >
    4
  </button>
  <button
    class="Pagerstyles__Box-sc-1weko6t-0 dWrcUk"
    tabindex="1"
  >
    <svg
      fill="currentColor"
      height="20"
      viewBox="0 0 24 24"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs />
      <path
        d="M14.44 11.854l-7.294 7.293 1.415 1.414 8-8a1 1 0 000-1.415l-8-8-1.415 1.415 7.293 7.293z"
      />
    </svg>
  </button>
</div>
`;

exports[`Pagination w/ size 3`] = `
<div
  class="Paginationstyles__Box-sc-7a3gdv-0 kCPgsd"
>
  <button
    class="Pagerstyles__Box-sc-1weko6t-0 gaebIT"
    disabled=""
    tabindex="-1"
  >
    <svg
      fill="currentColor"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs />
      <path
        d="M9.56 11.854l7.294 7.293-1.415 1.414-8-8a1 1 0 010-1.415l8-8 1.415 1.415-7.293 7.293z"
      />
    </svg>
  </button>
  <button
    class="Pagerstyles__Box-sc-1weko6t-0 kgWAfk"
    tabindex="1"
  >
    1
  </button>
  <button
    class="Pagerstyles__Box-sc-1weko6t-0 hDYgNB"
    tabindex="1"
  >
    2
  </button>
  <button
    class="Pagerstyles__Box-sc-1weko6t-0 hDYgNB"
    tabindex="1"
  >
    3
  </button>
  <button
    class="Pagerstyles__Box-sc-1weko6t-0 hDYgNB"
    tabindex="1"
  >
    4
  </button>
  <button
    class="Pagerstyles__Box-sc-1weko6t-0 hDYgNB"
    tabindex="1"
  >
    <svg
      fill="currentColor"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs />
      <path
        d="M14.44 11.854l-7.294 7.293 1.415 1.414 8-8a1 1 0 000-1.415l-8-8-1.415 1.415 7.293 7.293z"
      />
    </svg>
  </button>
</div>
`;
